processing/src/test/java/org/apache/druid/query/InlineDataSourceTest.java

```diff
@@ -225,6 +225,14 @@ public void test_equals()
                   .verify();
   }
 
+  @Test
+  public void test_toString_iterable()
+  {
+    // Verify that toString does not iterate the rows.
+    final String ignored = iterableDataSource.toString();
+    Assert.assertEquals(0, iterationCounter.get());
```

Comment:
I tried adding `rows` to `toString` and it doesn't iterate over `rowsIterable` (it prints an identifier for the lambda).

Reply:
Yeah, that makes sense. I guess the behavior would depend on whatever the particular Iterable’s toString method does. Do you think it makes sense to leave the test like this it change it somehow?

Reply:
I'm ok with leaving the test as it is for now.

Reply:
Any suggestions for how to improve it in a follow-up?